lib/api/login.js

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di'),
+    express = require('express');
+
+module.exports = loginRouterFactory;
+
+di.annotate(loginRouterFactory, new di.Provide('Http.Api.Login'));
+di.annotate(loginRouterFactory, new di.Inject(
+        'Auth.Services'
+    )
+);
+
+function loginRouterFactory (
+    authService
+) {
+
+
+    var router = express.Router();
+
+    /**
+     * @api {get} /login/ GET /
+     * @apiVersion 1.1.0
+     * @apiDescription get a token before accessing other apis
+     * @apiName login-get
+     * @apiGroup login
+     * @apiParam {query} username username used to login
+     * @apiParam {query} password password used to login
+     * @apiSuccess {json} a token to be used when accessing other apis:
+     *      {
+     *       "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyIjoiY
+     *       WRtaW4iLCJpYXQiOjE0NTI0ODA3MTIsImV4cCI6MTQ1MjU2NzExMn0.Qo-9y
+     *       omlkd0WE1jlEA-L1WqA6k_9qJYhkicQL5IUmSg"
+     *      }
+     */
+
+    router.get('/login', authService.authMiddlewareLogin.bind(authService));
```

Comment:
@cgx027 I don't understand why a GET handler is needed.  The semantics for the GET method ignore the request body, so the only way this can work is if username and password are provided as a query string in plain text.